### PR TITLE
Add SMART attributes for SSD/NVMe to "zpool status -c"

### PR DESCRIPTION
**Pull request by bunder2015**
### Description

This adds the SMART attributes required to probe Samsung SSD and NVMe
(and possibly others) disks when using the "zpool status -c" command.
### Motivation and Context

#7183 
### How Has This Been Tested?

Manual testing on pools consisting of SSD and NVMe
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux code style requirements.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain `Signed-off-by`.
- [x] Change has been approved by a ZFS on Linux member.

### CODE REVIEW

File: cmd/zpool/zpool.d/smart [hunk 1]
```diff
@@ -37,15 +37,18 @@ smartctl_path=$(which smartctl)
...
+	# What kind of drive are we?  Look for the right line in smartctl:
 	#
 	# SAS:
 	#	Transport protocol:   SAS
 	#
 	# SATA:
 	#	ATA Version is:   8
 	#
-	type=$(echo "$raw_out" | grep -m 1 -Eo '^ATA|SAS$')
+        # NVMe:
```
Comment (loli10K) [inline cmd/zpool/zpool.d/smart]: nit: replace spaces with an hard tab.

**Comment by bunder2015** [comment]
Oops.  I thought I fixed that before pushing.  Fixed.

File: cmd/zpool/zpool.d/smart [hunk 2]
```diff
@@ -71,10 +74,14 @@ if [ -b "$VDEV_UPATH" ] && [ -x "$smartctl_path" ]; then
 
 # SATA common
 /Temperature_Celsius/{print "temp="$10}
+/Airflow_Temperature_Cel/{print "temp="$10}
 /SMART overall-health self-assessment test result:/{print "health="$6}
 /Power_On_Hours/{print "hours_on="$10}
 /Serial Number:/{print "serial="$3}
 
+# NVMe specific
+/Temperature:/{print "temp="$2}
```
[the remark this answers is not in the document]
Comment (bunder2015) [reply cmd/zpool/zpool.d/smart]: I can add more if you would like.  I wasn't sure which other fields would be desirable.  Here is the full output from `smartctl -a` on an NVMe device: https://gist.github.com/bunder2015/95345e8e84d78f98eb3a35a8a29b58d6

Give me a few and I'll take another stab at what I have so far.

[the remark this answers is not in the document]
Comment (gdevenyi) [reply cmd/zpool/zpool.d/smart]: There's also the nvme-cli tool which accesses a bunch more info: https://github.com/linux-nvme/nvme-cli

**Comment by bunder2015** [comment]
@dinatale2 I decided to take your advice and added a few more attributes to bring it close in line with what we are using on SATA.  If you want me to add more, let me know.

File: cmd/zpool/zpool.d/nvme_err [hunk 3]
```diff
@@ -0,0 +1 @@
+smart
```
Comment (dinatale2) [inline cmd/zpool/zpool.d/nvme_err]: You're going to need to update `cmd/zpool/Makefile.am` to include this.

Comment (bunder2015) [reply cmd/zpool/zpool.d/nvme_err]: D'oh!  Fixed.

**Pull request merged**